h2o-parsers/h2o-parquet-parser/src/main/java/org/apache/parquet/hadoop/VecParquetReader.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+
+import org.apache.parquet.avro.AvroReadSupport;
+import org.apache.parquet.format.converter.ParquetMetadataConverter;
+import static org.apache.parquet.format.converter.ParquetMetadataConverter.MetadataFilter;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+import water.fvec.Vec;
+import water.parser.parquet.VecDataInputStream;
+import water.parser.parquet.VecFileSystem;
+import water.util.Log;
+
+import static org.apache.parquet.bytes.BytesUtils.readIntLittleEndian;
+import static org.apache.parquet.hadoop.ParquetFileWriter.MAGIC;
+
+/**
+ * Implementation of Parquet Reader working on H2O's Vecs.
+ *
+ * Note: This class was derived from Parquet's ParquetReader implementation. We cannot directly
+ * use the original implementation because it uses Hadoop FileSystem to access source data (and also Parquet summary files),
+ * it uses its own parallel implementation for reading metadata information which doesn't fit into H2O's architecture.
+ * We need to keep this class in package "org.apache.parquet.hadoop" to get access to Parquet's InternalParquetRecordReader.
+ */
+public class VecParquetReader implements Closeable {
+
+  private static ParquetMetadataConverter converter = new ParquetMetadataConverter();
```

Comment:
Is the `ParquetMetadataConverter` thread safe? E.g., if we invoke two parquet parse jobs